core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -95,6 +95,10 @@ public TableReader(CairoConfiguration configuration, CharSequence tableName) {
             LOG.info().$("table [id=").$(metadata.getId()).$(']').$();
             this.txFile = new TxReader(ff, path, partitionBy);
             readTxnSlow();
+            if (this.txn == TableUtils.INITIAL_TXN) {
```

Comment:
this already done in txnFile. `readTxnSlow()` could have this:

```java
            // exit if this is the same as we already have
            if (txn == this.txn) {
                txnScoreboard.acquireTxn(txn);
                if(txn == TableUtils.INITIAL_TXN) {
                    this.txFile.readSymbolCounts(this.symbolCountSnapshot);
                }
                return false;
            }

```

Reply:
I changed it without adding to execution in readTxnSlow by setting initial txn to -1 which forces reader to read tx file on init